packages/contracts-bedrock/src/L1/OptimismPortal2.sol

```diff
@@ -688,10 +727,68 @@ contract OptimismPortal2 is Initializable, ResourceMetering, ReinitializableBase
 
     /// @notice Migrates the total ETH balance to the ETHLockbox.
     function migrateLiquidity() public {
-        if (msg.sender != proxyAdminOwner()) revert OptimismPortal_Unauthorized();
+        if (msg.sender != proxyAdminOwner()) {
+            revert OptimismPortal_Unauthorized();
+        }
         _migrateLiquidity();
     }
 
+    /// @notice Add a minter to the OptimismPortal contract. To disable, set an empty value.
+    function setMinter(address _minter) external {
+        if (msg.sender != proxyAdminOwner()) {
+            revert OptimismPortal_Unauthorized();
+        }
+        QKCConfigStorage storage $ = _getQKCConfigStorage();
+        $.minter = _minter;
+        emit MinterSet(_minter);
+    }
+
+    /// @notice Mint a specific amount of L2 native token to an address.
+    function mintTransaction(address _to, uint256 _value) external metered(RECEIVE_DEFAULT_GAS_LIMIT) {
+        QKCConfigStorage storage $ = _getQKCConfigStorage();
+        if (msg.sender != $.minter) {
+            revert OptimismPortal_Unauthorized();
+        }
+
+        if (_to == address(0)) {
+            revert OptimismPortal_BadTarget();
+        }
+        // Compute the opaque data that will be emitted as part of the TransactionDeposited event.
+        // We use opaque data so that we can update the TransactionDeposited event in the future
+        // without breaking the current interface.
+        bytes memory opaqueData = abi.encodePacked(_value, _value, RECEIVE_DEFAULT_GAS_LIMIT, false, bytes(""));
+
+        // Emit a TransactionDeposited event so that the rollup node can derive a deposit
+        // transaction for this deposit.
+        emit TransactionDeposited(Constants.DEPOSITOR_ACCOUNT, _to, DEPOSIT_VERSION, opaqueData);
```

Comment:
We should change the DEPOSITOR_ACCOUNT to another fixed address to avoid potential risky calls.

Reply:
Good point, I also changed `setNativeDeposit` to use the depositor #2 .